### ISSUE TITLE
[RFC][V2] Support glob at package.json#source

### ISSUE THREAD

**Issue by brillout** [parcel-bundler/parcel#3821]
# 💬 RFC

Currently, `EntryResolver` expects `package.json#source` to be the path of a file.

I'm building an [SSR plugin for Parcel v2](https://github.com/brillout/parcel-ssr) and I'd like my users to be able to define multiple entry points. For example:

~~~
pages/
pages/package.json
pages/Landing.page.js
pages/about/About.page.js
pages/product/Product.page.js
...
~~~

where `pages/package.json#source` would be `**/*.page.js`. But globs are not supported for `package.json#source`.

It seems that currently the only way to have multiple entries is to have a `package.json` for each entry. But it would be silly to ask my users to create a new `package.json` each time they create a new page.

This is somewhat related to the discussion at #3302.

**Comment by ngdangtu-vn** [comment]
This definitely a problem I encounter recently. Does anyone know why this bug hasn't been fixed? It has been a year since then right? Btw, my problem can not be solved by listing all the possible entries in `source` because it is very dynamic. A new entries can be added and removed anytime. Put a fix path is not a very good solution.

I believe glob for `package.targets.*.source` is necessary.

**Comment by buvinghausen** [comment]
Let me just state this is a +1 issue for me too.

My use case is having parcel build all the LESS files across multiple services and output the CSS in the same folder as the source less file so that Blazor will apply CSS Isolation (CSS Modules to the rest of the world) correctly. I need the ability to set multiple targets each with their own source (as a glob) and distDir which cannot be done in a single CLI command as far as I can tell.

I guess as a workaround I can use a library like npm-run-all and run all the builds/watches in parallel it would just be nice to have glob support in package.json#targets.*.source and have one parcel thread figure out all the caching build/watch scenarios.

**Comment by buvinghausen** [comment]
@mischnic I see a ```Port "1234" could not be used``` message pop up in the console when I spin up parcel watch in parallel threads but overall it does seem to be working as it recompiles the LESS files into CSS files from both thread watches.

**Comment by mischnic** [comment]
Well, if you start multiple instances of Parcel, they can't all use the same port. 
Just like starting multiple webservers won't automatically expose all of them correctly on the same port

**Comment by dev-inside** [comment]
Hey,
I stumbled over this post a couple of days ago. As I see, there is still no support for using package.json#source with glob-entries? Any news on this issue? I would love to get rid of my 10km long scripts^^

**Comment by Darrenbydesign** [comment]
I too am curious bout this getting resolved... as a newb to bundling I was enticed by Parcel for building a component library with less fuss than something like WebPack which made me want to pull my hair out... but this glob support is the one roadblock to smooth building for me

**Comment by bAmrish** [comment]
I too am facing a problem where I need to specify multiple files in `package.json#source` section. Any information on if this bug is going to be fixed or not (either way) would be helpful.

**Comment by buvinghausen** [comment]
> I too am facing a problem where I need to specify multiple files in `package.json#source` section. Any information on if this bug is going to be fixed or not (either way) would be helpful.

It's been open for 4+ years I think you can infer it will not be supported on any reasonable timeline at this point.

**Comment by Darrenbydesign** [comment]
> > I too am facing a problem where I need to specify multiple files in `package.json#source` section. Any information on if this bug is going to be fixed or not (either way) would be helpful.
> 
> It's been open for 4+ years I think you can infer it will not be supported on any reasonable timeline at this point.

Sadly you're likely correct

**Comment by dev-inside** [comment]
Well, then the docs might need an update. The good guys from Parcel should mention how and where it can be used or get rid of the part in the documentation completely.
